File: Python/0451-0500/485. Max Consecutive Ones.py
# https://leetcode.com/problems/max-consecutive-ones/description/

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        oneCount, result = 0, 0

        for index in range(0, len(nums)):
            if nums[index] == 0:
                # update result
                result = max(result, oneCount)
                oneCount = 0
            else:
                oneCount += 1

        return max(result, oneCount)

File: Java/0451-0500/485. Max Consecutive Ones.java
// https://leetcode.com/problems/max-consecutive-ones/description/

class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int oneCount = 0, result = 0;

        for(int index=0; index<nums.length; index++){
            if(nums[index] == 0){
                // update result
                result = Math.max(result, oneCount);
                oneCount = 0;
            }else
                oneCount++;
        }
        //System.gc(); // optional to free-up memory

        return Math.max(result, oneCount);
    }
}
